### PR TITLE
THNETDIR and WIFINM: Align with latest spec and other small fixes

### LINKED ISSUE #359: [TC-THNETDIR-2.2] YAML script needs to be updated with AddNetwork and RemoveNetwork command related test steps.
## Description:
`TC-THNETDIR-2.2` test case YAML script needs to be updated with AddNetwork and RemoveNetwork command related test steps. As of now the script is not matching with the test plan.

## Test plan Reference:
https://github.com/CHIP-Specifications/chip-test-plans/blob/master/src/cluster/thread_network_directory.adoc#tc-thnetdir-2-2-adding-removing-and-reading-thread-networks-with-dut_server

## Script Reference: 
https://github.com/project-chip/connectedhomeip/blob/master/src/app/tests/suites/certification/Test_TC_THNETDIR_2_2.yaml

image

### PR DESCRIPTION
**Pull request by ksperling-apple**
Aligns implementation with https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/10295

- THNETDIR: Implement ActiveTimestamp check for updates
- THNETDIR: Implement CASE session requirement for GetOperationalDataset
- WIFINM: Implement CASE session requirement for NetworkPassphraseRequest
- Update / add certification YAML tests accordingly

Fixes https://github.com/project-chip/matter-test-scripts/issues/359

### CODE REVIEW

**Review by tleacmcsa — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by ksperling-apple** [comment]
Rebased to pick up https://github.com/project-chip/connectedhomeip/pull/35206

File: src/app/tests/suites/certification/Test_TC_THNETDIR_2_3.yaml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+              - name: PAKEPasscodeVerifier
+                value: PakeVerifier
+              - name: Discriminator
+                value: discriminator
+              - name: Iterations
+                value: 1000
+              - name: Salt
+                value: "SPAKE2P Key Salt"
+
+    - label: "TH establishes a PASE session with the DUT"
```
Comment (marchemi) [inline src/app/tests/suites/certification/Test_TC_THNETDIR_2_3.yaml:63]:
```suggestion
    - label: "TH2 establishes a PASE session with the DUT"
```

File: src/app/tests/suites/certification/Test_TC_THNETDIR_2_3.yaml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      command: EstablishPASESession
+      arguments:
+          values:
+              - name: nodeId
+                value: nodeId
+              - name: payload
+                value: payload
+
+    - label:
+          "TH sends GetOperationalDataset command to the DUT over the PASE
```
Comment (marchemi) [inline src/app/tests/suites/certification/Test_TC_THNETDIR_2_3.yaml:76]:
```suggestion
          "TH2 sends GetOperationalDataset command to the DUT over the PASE
```

File: src/app/tests/suites/certification/Test_TC_WIFINM_2_2.yaml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+              - name: PAKEPasscodeVerifier
+                value: PakeVerifier
+              - name: Discriminator
+                value: discriminator
+              - name: Iterations
+                value: 1000
+              - name: Salt
+                value: "SPAKE2P Key Salt"
+
+    - label: "TH establishes a PASE session with the DUT"
```
Comment (marchemi) [inline src/app/tests/suites/certification/Test_TC_WIFINM_2_2.yaml:64]:
```suggestion
    - label: "TH2 establishes a PASE session with the DUT"
```

File: src/app/tests/suites/certification/Test_TC_WIFINM_2_2.yaml [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+      command: EstablishPASESession
+      arguments:
+          values:
+              - name: nodeId
+                value: nodeId
+              - name: payload
+                value: payload
+
+    - label:
+          "TH sends the NetworkPassphraseRequest command to the DUT over the
```
Comment (marchemi) [inline src/app/tests/suites/certification/Test_TC_WIFINM_2_2.yaml:77]:
```suggestion
          "TH2 sends the NetworkPassphraseRequest command to the DUT over the
```

**Review by marchemi — APPROVED** [review]

**Review by jmartinez-silabs — APPROVED** [review]

**Pull request merged**